Xwt/Xwt/MessageDialog.cs

```diff
@@ -305,7 +305,8 @@ static Command GenericAlert (WindowFrame parent, MessageDescription message)
 			using (backend) {
 				Command res = null;
 				Toolkit.CurrentEngine.InvokePlatformCode (delegate {
-					res = backend.Run (parent ?? RootWindow, message);
+					WindowFrame parentWindow = parent ?? RootWindow ?? Toolkit.CurrentEngine.GetActiveWindow() as WindowFrame;
+					res = backend.Run (parentWindow, message);
```

Comment:
It sounds like something the toolkit could implement instead of having a GetActiveWindow() API.

Reply:
`IAlertDialogBackend` should be key/focused window aware

Reply:
In example, I think it's better to just implement it here:
https://github.com/mono/xwt/blob/83ebb134aa9457313bac90c786fc8201dfa7e1b5/Xwt.XamMac/Xwt.Mac/AlertDialogBackend.cs#L111

Reply:
Yeah, I was thinking on this yesterday, in fact in other toolkits this probably doesn't makes sense the parenting works in a different way. I moved the logic to